lib/system/sets.nim

```diff
@@ -30,6 +30,10 @@ proc countBits64(n: uint64): int {.compilerproc.} =
   result = ((v * 0x0101010101010101'u64) shr 56'u64).int
 
 proc cardSet(s: NimSet, len: int): int {.compilerproc, inline.} =
-  for i in 0..<len:
-    if likely(s[i] == 0): continue
+  var j = -1
+  for i in countup(0, len - 8, 8):
+    inc(result, countBits64((cast[ptr uint64](s[i].unsafeAddr))[]))
```

Comment:
This is very likely to become an unaligned pointer. In your example program you are lucky and the array will be in aligned memory. You should either ensure to start at an aligned memory address here, or the type itself should change to be an array of ``uint64`` to ensure a pointer to the beginning is 8 byte aligned.

Reply:
On x86 unaligned are no problem though. Maybe we should make this architecture specific.

Reply:
so:
```Nim
proc cardSet(s: NimSet, len: int): int {.compilerproc, inline.} =
  var j = -1
  when defined(x86):
    for i in countup(0, len - 8, 8):
      inc(result, countBits64((cast[ptr uint64](s[i].unsafeAddr))[]))
      j = i + 7

  for i in (j + 1) ..< len:
    inc(result, countBits32(uint32(s[i])))
```
?